lib/rules/no-dupe-disjunctions.ts

```diff
@@ -22,106 +630,178 @@ export default createRule("no-dupe-disjunctions", {
             {
                 type: "object",
                 properties: {
-                    disallowNeverMatch: { type: "boolean" },
```

Comment:
I want to leave `disallowNeverMatch` in the schema and remove it in v1.0. We don't have to handle `disallowNeverMatch`.

Reply:
Good idea. Should I log a warning if the user tries to use it?

Reply:
I don't think we need log. Users who used or did not use `disallowNeverMatch` will be able to use the enhanced `no-dupe-disjunctions` in the next minor version.
By removing the option from the schema in the v1.0 release, users who use the option in v1.0 will be warned.